src/wp-content/themes/twentytwentyone/assets/sass/07-utilities/color-palette.scss

```diff
@@ -1,42 +1,92 @@
 // Gutenberg text color options
 
 .has-black-color[class] {
+	// Localize CSS-variables to limit relationship scope
+	[class*="__inner-container"] {
+		--local--color-primary: var(--global--color-black, #000000);
```

Comment:
What is the benefit of setting the variable here, rather than just directly combining these lines like:
```color: var(--global--color-black, #000000);```

Reply:
@scruffian, so this is a scoping trick I’m using so that button styles can “be aware” of what nested color rules they should follow in various contexts. 

It’s kind of difficult to explain in text but the default button color needs to always be accessible when nested. By setting a “local variable” we’re able to set one variable thats restricted to the nested wrapper (`"__inner-container"`) and another that can tap into the global variables when we need to which pretty much guarantees a11y and produces less CSS overall. 

Without this trick, the text-color assigned by the `.has-black-color` will bleed into the color rules for buttons and force us to write ever-ending overrides to try to account for all the possible conditions. It gets really messy really quickly that way :-)